sdk/nodejs/serverless/function.ts

```diff
@@ -254,55 +270,51 @@ interface Package {
 
 // allFolders computes the set of package folders that are transitively required by the given list of package
 // dependencies rooted in a package at the provided path.
-function allFoldersForPackages(path: string, packages: string[]): Promise<Set<string>> {
+function allFoldersForPackages(includedPackages: Set<string>, excludedPackages: Set<string>): Promise<Set<string>> {
     return new Promise((resolve, reject) => {
-        readPackageTree(path, undefined, (err: any, root: Package) => {
+        readPackageTree(".", undefined, (err: any, root: Package) => {
             if (err) {
                 return reject(err);
             }
+
+            const allPackages = new Set<string>(includedPackages);
+            for (const depName of Object.keys(root.package.dependencies)) {
```

Comment:
Perhaps a comment on this - since it is really the crux of the new behavior that we root the list based on the explicit `dependencies` of the root package.  (And in particular, don't include `devDependencies` or `peerDependencies`).

Reply:
sure